app/src/main/java/com/amaze/filemanager/asynchronous/AbstractRepeatingRunnable.java

```diff
@@ -0,0 +1,36 @@
+package com.amaze.filemanager.asynchronous;
+
+import android.support.annotation.NonNull;
+
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+
+public abstract class AbstractRepeatingRunnable implements Runnable {
+
+    protected final ScheduledFuture handle;
+
+    public AbstractRepeatingRunnable(long initialDelay, long period, @NonNull TimeUnit unit, boolean startImmediately) {
+        if(!startImmediately) {
+            throw new UnsupportedOperationException("RepeatingRunnables are immediatly excecuted!");
+        }
+
+        ScheduledExecutorService threadExcecutor = Executors.newScheduledThreadPool(0);
+        handle = threadExcecutor.scheduleAtFixedRate(this, initialDelay, period, unit);
+    }
+
+    public abstract void run();
```

Comment:
It is not necessary to re-declare `public abstract void run()` here - you are implementing `Runnable`, and you expect `Runnable.run()` is implemented by subclasses of `AbstractRepeatingRunnable`.